torchvision/datasets/video_utils.py

```diff
@@ -255,6 +255,50 @@ def get_clip_location(self, idx):
             clip_idx = idx - self.cumulative_sizes[video_idx - 1]
         return video_idx, clip_idx
 
+    def get_idx_range_of_video(self, idx):
+        """
+        Provided an idx of a clip, the function returns the indices range belong to the same video
+        """
+
+        video_idx = bisect.bisect_right(self.cumulative_sizes, idx)
+        if video_idx == 0:
+            idx_start = 0
+        else:
+            idx_start = self.cumulative_sizes[video_idx - 1]
+
+        if len(self.cumulative_sizes) == 0:
+            idx_end = 0
+        else:
+            idx_end = self.cumulative_sizes[video_idx] - 1
+
+        return idx_start, idx_end
+
+    def get_video_path(self, idx):
+        """
+        Converts a flattened representation of the indices into a video path
+        """
+        video_idx = bisect.bisect_right(self.cumulative_sizes, idx)
+        return self.video_paths[video_idx]
+
+    def get_video_name(self, idx):
+        """
+        Converts a flattened representation of the indices into a video name
+        """
+        from os import path, sep
+        video_path = self.get_video_path(idx)
+        file_no_ext, ext = path.splitext(video_path)
+        video_name = file_no_ext.split(sep)[-1]
+        return video_name
+
+    def get_frames_range(self, idx):
+        """
+        Converts a flattened representation of the indices into the indices of the frames in the original video
+        """
+        video_idx, clip_idx = self.get_clip_location(idx)
+        start_frame = clip_idx * self.step
+        end_frame = start_frame + self.num_frames - 1
```

Comment:
Same comment here, I'm not sure that this is correct.
The `pts` for a specific clip can be obtained via https://github.com/pytorch/vision/blob/cd1748443afb1ccd168a3b3da29317ca5d4cd253/torchvision/datasets/video_utils.py#L248, and the start / end are in the first and last position

Reply:
I was probably confused with clip_pts and clip_idx...
What are their purposes then?

Reply:
`clip_pts` is the presentation timestamp for all the frames in a clip. `clip_idx` is the index on the possible number of clips we have